01_host/05_consensus/block_production.adoc

```diff
@@ -121,26 +128,91 @@ for epoch stem:[cc E_n] and stem:[c in (0, 1)] is the BABE constant
 ====
 
 [#algo-block-production-lottery]
-===== Block Production Lottery
+===== Primary Block Production Lottery
 ****
-A block producer aiming to produce a block during stem:[cc E_n] should run
-the Algorithm as defined in <<>> to identify the slots it is awarded. These are
-the slots during which the block producer is allowed to build a block. The
-session secret key, stem:[sk], is the block producer lottery secret key and
+A block producer aiming to produce a block during stem:[cc E_n] should run the
+stem:["Block-Production-Lottery"] algorithm to identify the slots it is awarded.
+These are the slots during which the block producer is allowed to build a block.
+The session secret key, stem:[sk], is the block producer lottery secret key and
 stem:[n] is the index of the epoch for whose slots the block producer is running
 the lottery.
 
 Algorithm: stem:["Block-Production-Lottery"(sk)]
 
 . stem:[r larr "Epoch-Randomness"(n)]
-. stem:["for " i := 1 " to " sc_n]
-. stem:["    " (pi, d) larr "VRF"(r,i,sk)]
-. stem:["    " A\[i\] larr (d, pi)]
+. stem:["for " s := 1 " to " sc_n]
+. stem:["    " (o, p) larr "VRF"(r,s,e_i,s_k, p_k)]
+. stem:["    " A\[i\] larr (o, p)]
 . stem:["return " A]
 
-For any slot stem:[i] in epoch stem:[n] where stem:[d < r], the block producer
-is required to produce a block. For the definitions of _Epoch-Randomness_
-(<<defn-epoch-randomness>>) and _VRF_ functions (<<sect-vrf>>).
+where stem:["Epoch-Randomness"] is defined in (<<defn-epoch-randomness>>),
+stem:[sc_n] is defined in <<defn-epoch-duration>> , stem:["VRF"] creates the
+BABE VRF transcript (<<defn-babe-vrf-transcript>>) and stem:[e_i] is the epoch
+index, retrieved from the Runtime (<<sect-rte-babeapi-epoch>>). stem:[s_k] and
+stem:[p_k] is the secret key respectively the public key of the authority. For
+any slot stem:[s] in epoch stem:[n] where stem:[d < r], the block producer is
+required to produce a block.
+
+Note that the secondary slots (<<block-production-secondary-slots>>) are running
+along side the primary block production lottery.
```

Comment:
I would probably say that it runs the secondary slot "as a fallback" for primary block production.